x-pack/plugin/sql/src/main/java/org/elasticsearch/xpack/sql/analysis/analyzer/PreAnalyzer.java

```diff
@@ -23,9 +24,9 @@
     public static class PreAnalysis {
         public static final PreAnalysis EMPTY = new PreAnalysis(emptyList());
 
-        public final List<TableIdentifier> indices;
+        public final List<Tuple<TableIdentifier, Boolean>> indices;
```

Comment:
TableIdentifier is mapped to parsing and has a well defined scope - frozen is a property of a table not of its id (which is just another property).
Since in the future we might have other properties (potentially remote from CCS), I've retrofitted the tuple into a `TableInfo` class inside the analysis package.